cfgov/v1/forms.py

```diff
@@ -314,3 +317,53 @@ def get_query_strings(self):
             'tags__name__in',        # topics
             'authors__name__in',     # authors
         ]
+
+
+class NewsroomFilterForm(FilterableListForm):
+
+    def get_pages(self, parent, hostname):
+        pages = CFGOVPage.objects.live_shared(hostname).descendant_of(parent)
+        blogs = []
+        try:
+            blog = CFGOVPage.objects.get(slug='blog')
+            blogs = CFGOVPage.objects.live_shared(hostname).descendant_of(blog)
+        except CFGOVPage.DoesNotExist:
+            print 'A blog landing page needs to be made'
+        if blogs:
+            pages = list(chain(pages, blogs))
+        return pages
+
+    # Populate Topics' choices
+    def set_topics(self, parent, hostname):
+        tags = [tag for tags in [page.tags.names() for page in self.get_pages(parent, hostname)] for tag in tags]
+        # Orders by most to least common tags
+        options = most_common(tags)
+        most = [(option, option) for option in options[:3]]
+        other = [(option, option) for option in options[3:]]
+        self.fields['topics'].choices = \
+            (('Most frequent', tuple(most)),
+             ('All other topics', tuple(other)))
+
+    # Populate Authors' choices
+    def set_authors(self, parent, hostname):
+        all_authors = [author for authors in
+                       [page.authors.names() for page in self.get_pages(parent, hostname)]
+                       for author in authors]
+        # Orders by most to least common authors
+        self.fields['authors'].choices = [(author, author) for author in
+                                          most_common(all_authors)]
+
+
+class ActivityLogFilterForm(NewsroomFilterForm):
+    def per_page_limit(self):
+        return 100
+
+    def get_pages(self, parent, hostname):
+        pages = {'blog': None, 'newsroom': None, 'research-reports': None}
+        for slug in pages.keys():
+            try:
+                page = CFGOVPage.objects.get(slug=slug)
+                pages[slug] = CFGOVPage.objects.live_shared(hostname).descendant_of(page)
+            except CFGOVPage.DoesNotExist:
+                print slug, 'does not exist'
```

Comment:
If there aren't pages that have the slugs `blog`, `newsroom`, or `research-reports`.